pkg/apis/core/validation/validation.go

```diff
@@ -3758,6 +3758,10 @@ func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolic
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("searches"), dnsConfig.Searches, fmt.Sprintf("must not have more than %v characters (including spaces) in the search list", MaxDNSSearchListChars)))
 		}
 		for i, search := range dnsConfig.Searches {
+			// single dot is allowed #125883
```

Comment:
if we relax validation now, and a pod with the dot is stored, an apiserver without this option will fail to validate it and error